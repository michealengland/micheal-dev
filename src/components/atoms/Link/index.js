import React from 'react'
import { Link as GatsbyLink } from 'gatsby'
import PropTypes from 'prop-types'

export default function Link({ children, to, activeClassName, partiallyActive, ...other }) {
  const internal = /^\/(?!\/)/.test(to)

  // Use Gatsby Link for internal links, and <a> for others
  if (internal) {
    return (
      <GatsbyLink
        activeClassName={activeClassName}
        partiallyActive={partiallyActive}
        to={to}
        {...other}
      >
        {children}
      </GatsbyLink>
    )
  } else {
    return (
      <a href={to} {...other}>
        {children}
      </a>
    )
  }
}

Link.propTypes = {
  children: PropTypes.node.isRequired,
  partiallyActive: PropTypes.bool,
  to: PropTypes.string.isRequired,
}

Link.defaultProps = {
  displayType: false,
}
