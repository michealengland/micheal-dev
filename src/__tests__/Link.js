import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../components/atoms/Link'
describe('Link', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Link to="/">Home Link</Link>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
