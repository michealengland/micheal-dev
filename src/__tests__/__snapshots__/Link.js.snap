// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  displayType={false}
  href="/"
>
  Home Link
</a>
`;
